Cannot build on 32bit arm (ARMv6)

## Description

### Current State

Cannot build on ARMv6 with musl

### Expected

Expected to be able to build on ARMv6 with musl

## Environment

- Hardware Architecture: arm
- Operating system: alpine linux

Have tried the following combinations:

alpine 3.16 / gcc 11
```
- C++ Compiler version: gcc 11.2.1
- CMake version: 3.23.1
- CMake flags: '-DCMAKE_BUILD_TYPE=Release' '-DWASMEDGE_BUILD_STATIC_LIB=ON' '-DWASMEDGE_BUILD_SHARED_LIB=OFF' '-DWASMEDGE_BUILD_AOT_RUNTIME=OFF' '-DWASMEDGE_BUILD_TOOLS=OFF' -Wno-dev
- Boost version: 1.78
```

alpine 3.17 / gcc 12
```
- C++ Compiler version: gcc 12.2.1
- CMake version: 3.24.3
- CMake flags: '-DCMAKE_BUILD_TYPE=Release' '-DWASMEDGE_BUILD_STATIC_LIB=ON' '-DWASMEDGE_BUILD_SHARED_LIB=OFF' '-DWASMEDGE_BUILD_AOT_RUNTIME=OFF' '-DWASMEDGE_BUILD_TOOLS=OFF' -Wno-dev
- Boost version: 1.80
```

alpine 3.17 / clang 15
```
- C++ Compiler version: clang 15.0.7
- CMake version: 3.24.3
- CMake flags: '-DCMAKE_BUILD_TYPE=Release' '-DWASMEDGE_BUILD_STATIC_LIB=ON' '-DWASMEDGE_BUILD_SHARED_LIB=OFF' '-DWASMEDGE_BUILD_AOT_RUNTIME=OFF' '-DWASMEDGE_BUILD_TOOLS=OFF' -Wno-dev
- Boost version: 1.80
```

## Steps to Reproduce

1. configure as described above
2. make


I am coming here from the K3s project (https://github.com/k3s-io/k3s/pull/6872); attempting to build wasmedge for embedding in K3s. For portability reasons, we build all our shipping binaries statically linked against musl libc. This seems to be working fine on amd64 and aarch64, and I have given up on s390x as Rust doesn't have a target for musl on that platform, but I am stuck on 32bit arm.

Build logs for my various attempts are available at: https://drone-pr.k3s.io/k3s-io/runwasi



<details>
  <summary><b>gcc build failure output</b></summary>
  
```
 33%] Building CXX object lib/plugin/CMakeFiles/wasmedgePlugin.dir/plugin.cpp.o
In file included from /drone/src/WasmEdge-0.11.2/include/runtime/instance/module.h:24,
                 from /drone/src/WasmEdge-0.11.2/include/plugin/plugin.h:21,
                 from /drone/src/WasmEdge-0.11.2/lib/plugin/plugin.cpp:4:
/drone/src/WasmEdge-0.11.2/include/runtime/instance/table.h: In member function 'WasmEdge::Expect<void> WasmEdge::Runtime::Instance::TableInstance::setRefAddr(uint32_t, WasmEdge::RefVariant)':
/drone/src/WasmEdge-0.11.2/include/runtime/instance/table.h:146:17: error: use of deleted function 'constexpr WasmEdge::Variant<Types>& WasmEdge::Variant<Types>::operator=(const WasmEdge::Variant<Types>&) [with Types = {WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef}]'
  146 |     Refs[Idx] = Val;
      |                 ^~~
In file included from /drone/src/WasmEdge-0.11.2/include/common/types.h:20,
                 from /drone/src/WasmEdge-0.11.2/include/ast/type.h:19,
                 from /drone/src/WasmEdge-0.11.2/include/ast/description.h:17,
                 from /drone/src/WasmEdge-0.11.2/include/ast/section.h:16,
                 from /drone/src/WasmEdge-0.11.2/include/loader/shared_library.h:17,
                 from /drone/src/WasmEdge-0.11.2/include/plugin/plugin.h:19:
/drone/src/WasmEdge-0.11.2/include/common/variant.h:162:22: note: 'constexpr WasmEdge::Variant<Types>& WasmEdge::Variant<Types>::operator=(const WasmEdge::Variant<Types>&) [with Types = {WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef}]' is implicitly deleted because the default definition would be ill-formed:
  162 |   constexpr Variant &operator=(const Variant &) noexcept = default;
      |                      ^~~~~~~~
/drone/src/WasmEdge-0.11.2/include/common/variant.h:162:22: error: use of deleted function 'constexpr WasmEdge::VariadicUnion<FirstT, RestT ...>& WasmEdge::VariadicUnion<FirstT, RestT ...>::operator=(const WasmEdge::VariadicUnion<FirstT, RestT ...>&) [with FirstT = WasmEdge::UnknownRef; RestT = {WasmEdge::FuncRef, WasmEdge::ExternRef}]'
/drone/src/WasmEdge-0.11.2/include/common/variant.h:33:28: note: 'constexpr WasmEdge::VariadicUnion<FirstT, RestT ...>& WasmEdge::VariadicUnion<FirstT, RestT ...>::operator=(const WasmEdge::VariadicUnion<FirstT, RestT ...>&) [with FirstT = WasmEdge::UnknownRef; RestT = {WasmEdge::FuncRef, WasmEdge::ExternRef}]' is implicitly deleted because the default definition would be ill-formed:
   33 |   constexpr VariadicUnion &operator=(const VariadicUnion &) = default;
      |                            ^~~~~~~~
/drone/src/WasmEdge-0.11.2/include/common/variant.h:33:28: error: use of deleted function 'constexpr WasmEdge::VariadicUnion<FirstT, RestT ...>& WasmEdge::VariadicUnion<FirstT, RestT ...>::operator=(const WasmEdge::VariadicUnion<FirstT, RestT ...>&) [with FirstT = WasmEdge::FuncRef; RestT = {WasmEdge::ExternRef}]'
/drone/src/WasmEdge-0.11.2/include/common/variant.h:33:28: note: 'constexpr WasmEdge::VariadicUnion<FirstT, RestT ...>& WasmEdge::VariadicUnion<FirstT, RestT ...>::operator=(const WasmEdge::VariadicUnion<FirstT, RestT ...>&) [with FirstT = WasmEdge::FuncRef; RestT = {WasmEdge::ExternRef}]' is implicitly deleted because the default definition would be ill-formed:
/drone/src/WasmEdge-0.11.2/include/common/variant.h:33:28: error: use of deleted function 'WasmEdge::FuncRef& WasmEdge::FuncRef::operator=(const WasmEdge::FuncRef&)'
/drone/src/WasmEdge-0.11.2/include/common/types.h:63:8: note: 'WasmEdge::FuncRef& WasmEdge::FuncRef::operator=(const WasmEdge::FuncRef&)' is implicitly deleted because the default definition would be ill-formed:
   63 | struct FuncRef {
      |        ^~~~~~~
/drone/src/WasmEdge-0.11.2/include/common/types.h:63:8: error: non-static const member 'const uint32_t WasmEdge::FuncRef::Padding', cannot use default assignment operator
/drone/src/WasmEdge-0.11.2/include/common/variant.h:33:28: error: use of deleted function 'constexpr WasmEdge::VariadicUnion<FirstT, RestT ...>& WasmEdge::VariadicUnion<FirstT, RestT ...>::operator=(const WasmEdge::VariadicUnion<FirstT, RestT ...>&) [with FirstT = WasmEdge::ExternRef; RestT = {}]'
   33 |   constexpr VariadicUnion &operator=(const VariadicUnion &) = default;
      |                            ^~~~~~~~
/drone/src/WasmEdge-0.11.2/include/common/variant.h:33:28: note: 'constexpr WasmEdge::VariadicUnion<FirstT, RestT ...>& WasmEdge::VariadicUnion<FirstT, RestT ...>::operator=(const WasmEdge::VariadicUnion<FirstT, RestT ...>&) [with FirstT = WasmEdge::ExternRef; RestT = {}]' is implicitly deleted because the default definition would be ill-formed:
/drone/src/WasmEdge-0.11.2/include/common/variant.h:33:28: error: use of deleted function 'WasmEdge::ExternRef& WasmEdge::ExternRef::operator=(const WasmEdge::ExternRef&)'
/drone/src/WasmEdge-0.11.2/include/common/types.h:73:8: note: 'WasmEdge::ExternRef& WasmEdge::ExternRef::operator=(const WasmEdge::ExternRef&)' is implicitly deleted because the default definition would be ill-formed:
   73 | struct ExternRef {
      |        ^~~~~~~~~
/drone/src/WasmEdge-0.11.2/include/common/types.h:73:8: error: non-static const member 'const uint32_t WasmEdge::ExternRef::Padding', cannot use default assignment operator
In file included from /usr/include/c++/12.2.1/string:50,
                 from /usr/include/c++/12.2.1/stdexcept:39,
                 from /usr/include/c++/12.2.1/system_error:41,
                 from /usr/include/c++/12.2.1/bits/fs_fwd.h:35,
                 from /usr/include/c++/12.2.1/filesystem:44,
                 from /drone/src/WasmEdge-0.11.2/include/common/filesystem.h:50,
                 from /drone/src/WasmEdge-0.11.2/include/plugin/plugin.h:17:
/usr/include/c++/12.2.1/bits/stl_algobase.h: In instantiation of 'typename __gnu_cxx::__enable_if<(! std::__is_scalar<_Tp>::__value), void>::__type std::__fill_a1(_ForwardIterator, _ForwardIterator, const _Tp&) [with _ForwardIterator = WasmEdge::Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef>*; _Tp = WasmEdge::Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef>; typename __gnu_cxx::__enable_if<(! __is_scalar<_Tp>::__value), void>::__type = void]':
/usr/include/c++/12.2.1/bits/stl_algobase.h:951:21:   required from 'void std::__fill_a1(__gnu_cxx::__normal_iterator<_Iterator, _Container>, __gnu_cxx::__normal_iterator<_Iterator, _Container>, const _Tp&) [with _Ite = WasmEdge::Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef>*; _Cont = vector<WasmEdge::Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef> >; _Tp = WasmEdge::Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef>]'
/usr/include/c++/12.2.1/bits/stl_algobase.h:968:21:   required from 'void std::__fill_a(_FIte, _FIte, const _Tp&) [with _FIte = __gnu_cxx::__normal_iterator<WasmEdge::Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef>*, vector<WasmEdge::Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef> > >; _Tp = WasmEdge::Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef>]'
/usr/include/c++/12.2.1/bits/stl_algobase.h:1119:20:   required from '_OutputIterator std::__fill_n_a(_OutputIterator, _Size, const _Tp&, random_access_iterator_tag) [with _OutputIterator = __gnu_cxx::__normal_iterator<WasmEdge::Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef>*, vector<WasmEdge::Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef> > >; _Size = unsigned int; _Tp = WasmEdge::Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef>]'
/usr/include/c++/12.2.1/bits/stl_algobase.h:1148:29:   required from '_OI std::fill_n(_OI, _Size, const _Tp&) [with _OI = __gnu_cxx::__normal_iterator<WasmEdge::Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef>*, vector<WasmEdge::Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef> > >; _Size = unsigned int; _Tp = WasmEdge::Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef>]'
/drone/src/WasmEdge-0.11.2/include/runtime/instance/table.h:69:16:   required from here
/usr/include/c++/12.2.1/bits/stl_algobase.h:910:18: error: use of deleted function 'constexpr WasmEdge::Variant<Types>& WasmEdge::Variant<Types>::operator=(const WasmEdge::Variant<Types>&) [with Types = {WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef}]'
  910 |         *__first = __value;
      |         ~~~~~~~~~^~~~~~~~~
/usr/include/c++/12.2.1/bits/stl_algobase.h: In instantiation of 'static _Tp* std::__copy_move<_IsMove, true, std::random_access_iterator_tag>::__copy_m(const _Tp*, const _Tp*, _Tp*) [with _Tp = WasmEdge::Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef>; bool _IsMove = false]':
/usr/include/c++/12.2.1/bits/stl_algobase.h:495:30:   required from '_OI std::__copy_move_a2(_II, _II, _OI) [with bool _IsMove = false; _II = const WasmEdge::Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef>*; _OI = WasmEdge::Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef>*]'
/usr/include/c++/12.2.1/bits/stl_algobase.h:522:42:   required from '_OI std::__copy_move_a1(_II, _II, _OI) [with bool _IsMove = false; _II = const WasmEdge::Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef>*; _OI = WasmEdge::Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef>*]'
/usr/include/c++/12.2.1/bits/stl_algobase.h:530:31:   required from '_OI std::__copy_move_a(_II, _II, _OI) [with bool _IsMove = false; _II = const WasmEdge::Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef>*; _OI = __gnu_cxx::__normal_iterator<WasmEdge::Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef>*, vector<WasmEdge::Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef> > >]'
/usr/include/c++/12.2.1/bits/stl_algobase.h:620:7:   required from '_OI std::copy(_II, _II, _OI) [with _II = const WasmEdge::Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef>*; _OI = __gnu_cxx::__normal_iterator<WasmEdge::Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef>*, vector<WasmEdge::Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef> > >]'
/usr/include/c++/12.2.1/bits/stl_algo.h:728:23:   required from '_OutputIterator std::__copy_n(_RandomAccessIterator, _Size, _OutputIterator, random_access_iterator_tag) [with _RandomAccessIterator = const WasmEdge::Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef>*; _Size = unsigned int; _OutputIterator = __gnu_cxx::__normal_iterator<WasmEdge::Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef>*, vector<WasmEdge::Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef> > >]'
/usr/include/c++/12.2.1/bits/stl_algo.h:760:27:   required from '_OIter std::copy_n(_IIter, _Size, _OIter) [with _IIter = const WasmEdge::Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef>*; _Size = unsigned int; _OIter = __gnu_cxx::__normal_iterator<WasmEdge::Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef>*, vector<WasmEdge::Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef> > >]'
/drone/src/WasmEdge-0.11.2/include/runtime/instance/table.h:110:16:   required from here
/usr/include/c++/12.2.1/bits/stl_algobase.h:427:40: error: static assertion failed: type must be assignable
  427 |           static_assert( __assignable::value, "type must be assignable" );
      |                                        ^~~~~
/usr/include/c++/12.2.1/bits/stl_algobase.h:427:40: note: 'std::integral_constant<bool, false>::value' evaluates to false
make[2]: *** [lib/plugin/CMakeFiles/wasmedgePlugin.dir/build.make:76: lib/plugin/CMakeFiles/wasmedgePlugin.dir/plugin.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:466: lib/plugin/CMakeFiles/wasmedgePlugin.dir/all] Error 2
```
</details>

<details>
  <summary><b>clang build failure output</b></summary>

```
[ 33%] Building CXX object lib/plugin/CMakeFiles/wasmedgePlugin.dir/plugin.cpp.o
cd /drone/src/WasmEdge-0.11.2/lib/plugin && /usr/bin/clang++ -DSPDLOG_COMPILED_LIB -I/drone/src/WasmEdge-0.11.2/include/api -I/drone/src/WasmEdge-0.11.2/include -I/drone/src/WasmEdge-0.11.2/_deps/spdlog-src/include -I/drone/src/WasmEdge-0.11.2/lib/system -O3 -DNDEBUG -fPIC -fvisibility=hidden -fvisibility-inlines-hidden -Wall -Wextra -Werror -Wno-error=pedantic -Wno-psabi -Wno-c++20-designator -Wno-c99-extensions -Wno-covered-switch-default -Wno-documentation-unknown-command -Wno-error=nested-anon-types -Wno-error=old-style-cast -Wno-error=unused-command-line-argument -Wno-error=unknown-warning-option -Wno-ctad-maybe-unsupported -Wno-gnu-anonymous-struct -Wno-keyword-macro -Wno-language-extension-token -Wno-newline-eof -Wno-shadow-field-in-constructor -Wno-signed-enum-bitfield -Wno-switch-enum -Wno-undefined-func-template -Wno-unused-const-variable -Wno-error=shadow-field -Wno-reserved-identifier -std=c++17 -MD -MT lib/plugin/CMakeFiles/wasmedgePlugin.dir/plugin.cpp.o -MF CMakeFiles/wasmedgePlugin.dir/plugin.cpp.o.d -o CMakeFiles/wasmedgePlugin.dir/plugin.cpp.o -c /drone/src/WasmEdge-0.11.2/lib/plugin/plugin.cpp
In file included from /drone/src/WasmEdge-0.11.2/lib/plugin/plugin.cpp:4:
In file included from /drone/src/WasmEdge-0.11.2/include/plugin/plugin.h:21:
In file included from /drone/src/WasmEdge-0.11.2/include/runtime/instance/module.h:24:
/drone/src/WasmEdge-0.11.2/include/runtime/instance/table.h:146:15: error: object of type '__gnu_cxx::__alloc_traits<std::allocator<WasmEdge::Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef>>, WasmEdge::Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef>>::value_type' (aka 'WasmEdge::Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef>') cannot be assigned because its copy assignment operator is implicitly deleted
    Refs[Idx] = Val;
              ^
/drone/src/WasmEdge-0.11.2/include/common/variant.h:162:22: note: explicitly defaulted function was implicitly deleted here
  constexpr Variant &operator=(const Variant &) noexcept = default;
                     ^
/drone/src/WasmEdge-0.11.2/include/common/variant.h:156:27: note: copy assignment operator of 'Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef>' is implicitly deleted because field 'Storage' has a deleted copy assignment operator
  VariadicUnion<Types...> Storage;
                          ^
/drone/src/WasmEdge-0.11.2/include/common/variant.h:33:28: note: explicitly defaulted function was implicitly deleted here
  constexpr VariadicUnion &operator=(const VariadicUnion &) = default;
                           ^
/drone/src/WasmEdge-0.11.2/include/common/variant.h:94:27: note: copy assignment operator of 'VariadicUnion<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef>' is implicitly deleted because field 'Rest' has a deleted copy assignment operator
  VariadicUnion<RestT...> Rest;
                          ^
/drone/src/WasmEdge-0.11.2/include/common/variant.h:33:28: note: explicitly defaulted function was implicitly deleted here
  constexpr VariadicUnion &operator=(const VariadicUnion &) = default;
                           ^
/drone/src/WasmEdge-0.11.2/include/common/variant.h:93:10: note: copy assignment operator of 'VariadicUnion<WasmEdge::FuncRef, WasmEdge::ExternRef>' is implicitly deleted because field 'First' has a deleted copy assignment operator
  FirstT First;
         ^
/drone/src/WasmEdge-0.11.2/include/common/types.h:65:18: note: copy assignment operator of 'FuncRef' is implicitly deleted because field 'Padding' is of const-qualified type 'const uint32_t' (aka 'const unsigned int')
  const uint32_t Padding = -1;
                 ^
In file included from /drone/src/WasmEdge-0.11.2/lib/plugin/plugin.cpp:4:
In file included from /drone/src/WasmEdge-0.11.2/include/plugin/plugin.h:17:
In file included from /drone/src/WasmEdge-0.11.2/include/common/filesystem.h:50:
In file included from /usr/bin/../lib/gcc/armv7-alpine-linux-musleabihf/12.2.1/../../../../include/c++/12.2.1/filesystem:44:
In file included from /usr/bin/../lib/gcc/armv7-alpine-linux-musleabihf/12.2.1/../../../../include/c++/12.2.1/bits/fs_fwd.h:35:
In file included from /usr/bin/../lib/gcc/armv7-alpine-linux-musleabihf/12.2.1/../../../../include/c++/12.2.1/system_error:41:
In file included from /usr/bin/../lib/gcc/armv7-alpine-linux-musleabihf/12.2.1/../../../../include/c++/12.2.1/stdexcept:39:
In file included from /usr/bin/../lib/gcc/armv7-alpine-linux-musleabihf/12.2.1/../../../../include/c++/12.2.1/string:50:
/usr/bin/../lib/gcc/armv7-alpine-linux-musleabihf/12.2.1/../../../../include/c++/12.2.1/bits/stl_algobase.h:910:11: error: object of type 'WasmEdge::Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef>' cannot be assigned because its copy assignment operator is implicitly deleted
        *__first = __value;
                 ^
/usr/bin/../lib/gcc/armv7-alpine-linux-musleabihf/12.2.1/../../../../include/c++/12.2.1/bits/stl_algobase.h:951:12: note: in instantiation of function template specialization 'std::__fill_a1<WasmEdge::Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef> *, WasmEdge::Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef>>' requested here
    { std::__fill_a1(__first.base(), __last.base(), __value); }
           ^
/usr/bin/../lib/gcc/armv7-alpine-linux-musleabihf/12.2.1/../../../../include/c++/12.2.1/bits/stl_algobase.h:968:12: note: in instantiation of function template specialization 'std::__fill_a1<WasmEdge::Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef> *, std::vector<WasmEdge::Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef>>, WasmEdge::Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef>>' requested here
    { std::__fill_a1(__first, __last, __value); }
           ^
/usr/bin/../lib/gcc/armv7-alpine-linux-musleabihf/12.2.1/../../../../include/c++/12.2.1/bits/stl_algobase.h:1119:12: note: in instantiation of function template specialization 'std::__fill_a<__gnu_cxx::__normal_iterator<WasmEdge::Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef> *, std::vector<WasmEdge::Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef>>>, WasmEdge::Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef>>' requested here
      std::__fill_a(__first, __first + __n, __value);
           ^
/usr/bin/../lib/gcc/armv7-alpine-linux-musleabihf/12.2.1/../../../../include/c++/12.2.1/bits/stl_algobase.h:1148:19: note: in instantiation of function template specialization 'std::__fill_n_a<__gnu_cxx::__normal_iterator<WasmEdge::Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef> *, std::vector<WasmEdge::Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef>>>, unsigned int, WasmEdge::Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef>>' requested here
      return std::__fill_n_a(__first, std::__size_to_integer(__n), __value,
                  ^
/drone/src/WasmEdge-0.11.2/include/runtime/instance/table.h:69:10: note: in instantiation of function template specialization 'std::fill_n<__gnu_cxx::__normal_iterator<WasmEdge::Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef> *, std::vector<WasmEdge::Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef>>>, unsigned int, WasmEdge::Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef>>' requested here
    std::fill_n(Refs.end() - Count, Count, Val);
         ^
/drone/src/WasmEdge-0.11.2/include/common/variant.h:162:22: note: explicitly defaulted function was implicitly deleted here
  constexpr Variant &operator=(const Variant &) noexcept = default;
                     ^
/drone/src/WasmEdge-0.11.2/include/common/variant.h:156:27: note: copy assignment operator of 'Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef>' is implicitly deleted because field 'Storage' has a deleted copy assignment operator
  VariadicUnion<Types...> Storage;
                          ^
/drone/src/WasmEdge-0.11.2/include/common/variant.h:33:28: note: explicitly defaulted function was implicitly deleted here
  constexpr VariadicUnion &operator=(const VariadicUnion &) = default;
                           ^
/drone/src/WasmEdge-0.11.2/include/common/variant.h:94:27: note: copy assignment operator of 'VariadicUnion<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef>' is implicitly deleted because field 'Rest' has a deleted copy assignment operator
  VariadicUnion<RestT...> Rest;
                          ^
/drone/src/WasmEdge-0.11.2/include/common/variant.h:33:28: note: explicitly defaulted function was implicitly deleted here
  constexpr VariadicUnion &operator=(const VariadicUnion &) = default;
                           ^
/drone/src/WasmEdge-0.11.2/include/common/variant.h:93:10: note: copy assignment operator of 'VariadicUnion<WasmEdge::FuncRef, WasmEdge::ExternRef>' is implicitly deleted because field 'First' has a deleted copy assignment operator
  FirstT First;
         ^
/drone/src/WasmEdge-0.11.2/include/common/types.h:65:18: note: copy assignment operator of 'FuncRef' is implicitly deleted because field 'Padding' is of const-qualified type 'const uint32_t' (aka 'const unsigned int')
  const uint32_t Padding = -1;
                 ^
In file included from /drone/src/WasmEdge-0.11.2/lib/plugin/plugin.cpp:4:
In file included from /drone/src/WasmEdge-0.11.2/include/plugin/plugin.h:17:
In file included from /drone/src/WasmEdge-0.11.2/include/common/filesystem.h:50:
In file included from /usr/bin/../lib/gcc/armv7-alpine-linux-musleabihf/12.2.1/../../../../include/c++/12.2.1/filesystem:44:
In file included from /usr/bin/../lib/gcc/armv7-alpine-linux-musleabihf/12.2.1/../../../../include/c++/12.2.1/bits/fs_fwd.h:35:
In file included from /usr/bin/../lib/gcc/armv7-alpine-linux-musleabihf/12.2.1/../../../../include/c++/12.2.1/system_error:41:
In file included from /usr/bin/../lib/gcc/armv7-alpine-linux-musleabihf/12.2.1/../../../../include/c++/12.2.1/stdexcept:39:
In file included from /usr/bin/../lib/gcc/armv7-alpine-linux-musleabihf/12.2.1/../../../../include/c++/12.2.1/string:50:
/usr/bin/../lib/gcc/armv7-alpine-linux-musleabihf/12.2.1/../../../../include/c++/12.2.1/bits/stl_algobase.h:427:4: error: static assertion failed due to requirement 'is_copy_assignable<WasmEdge::Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef>>::value': type must be assignable
          static_assert( __assignable::value, "type must be assignable" );
          ^              ~~~~~~~~~~~~~~~~~~~
/usr/bin/../lib/gcc/armv7-alpine-linux-musleabihf/12.2.1/../../../../include/c++/12.2.1/bits/stl_algobase.h:495:22: note: in instantiation of function template specialization 'std::__copy_move<false, true, std::random_access_iterator_tag>::__copy_m<WasmEdge::Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef>>' requested here
                              _Category>::__copy_m(__first, __last, __result);
                                          ^
/usr/bin/../lib/gcc/armv7-alpine-linux-musleabihf/12.2.1/../../../../include/c++/12.2.1/bits/stl_algobase.h:522:19: note: in instantiation of function template specialization 'std::__copy_move_a2<false, const WasmEdge::Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef> *, WasmEdge::Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef> *>' requested here
    { return std::__copy_move_a2<_IsMove>(__first, __last, __result); }
                  ^
/usr/bin/../lib/gcc/armv7-alpine-linux-musleabihf/12.2.1/../../../../include/c++/12.2.1/bits/stl_algobase.h:530:8: note: in instantiation of function template specialization 'std::__copy_move_a1<false, const WasmEdge::Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef> *, WasmEdge::Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef> *>' requested here
                std::__copy_move_a1<_IsMove>(std::__niter_base(__first),
                     ^
/usr/bin/../lib/gcc/armv7-alpine-linux-musleabihf/12.2.1/../../../../include/c++/12.2.1/bits/stl_algobase.h:619:19: note: in instantiation of function template specialization 'std::__copy_move_a<false, const WasmEdge::Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef> *, __gnu_cxx::__normal_iterator<WasmEdge::Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef> *, std::vector<WasmEdge::Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef>>>>' requested here
      return std::__copy_move_a<__is_move_iterator<_II>::__value>
                  ^
/usr/bin/../lib/gcc/armv7-alpine-linux-musleabihf/12.2.1/../../../../include/c++/12.2.1/bits/stl_algo.h:728:19: note: in instantiation of function template specialization 'std::copy<const WasmEdge::Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef> *, __gnu_cxx::__normal_iterator<WasmEdge::Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef> *, std::vector<WasmEdge::Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef>>>>' requested here
    { return std::copy(__first, __first + __n, __result); }
                  ^
/usr/bin/../lib/gcc/armv7-alpine-linux-musleabihf/12.2.1/../../../../include/c++/12.2.1/bits/stl_algo.h:760:19: note: in instantiation of function template specialization 'std::__copy_n<const WasmEdge::Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef> *, unsigned int, __gnu_cxx::__normal_iterator<WasmEdge::Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef> *, std::vector<WasmEdge::Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef>>>>' requested here
      return std::__copy_n(__first, __n2, __result,
                  ^
/drone/src/WasmEdge-0.11.2/include/runtime/instance/table.h:110:10: note: in instantiation of function template specialization 'std::copy_n<const WasmEdge::Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef> *, unsigned int, __gnu_cxx::__normal_iterator<WasmEdge::Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef> *, std::vector<WasmEdge::Variant<WasmEdge::UnknownRef, WasmEdge::FuncRef, WasmEdge::ExternRef>>>>' requested here
    std::copy_n(Slice.begin() + Start, Length, Refs.begin() + Offset);
         ^
3 errors generated.
make[2]: *** [lib/plugin/CMakeFiles/wasmedgePlugin.dir/build.make:76: lib/plugin/CMakeFiles/wasmedgePlugin.dir/plugin.cpp.o] Error 1
```
</details>

I will also note when using cmake I had to add `-Wno-unused-const-variable` to get around the following error:
```
[ 18%] Building CXX object lib/system/CMakeFiles/wasmedgeSystem.dir/allocator.cpp.o
/drone/src/WasmEdge-0.11.2/lib/system/allocator.cpp:55:40: error: unused variable 'k4G' [-Werror,-Wunused-const-variable]
static inline constexpr const uint64_t k4G = UINT64_C(0x100000000);
                                       ^
/drone/src/WasmEdge-0.11.2/lib/system/allocator.cpp:56:40: error: unused variable 'k12G' [-Werror,-Wunused-const-variable]
static inline constexpr const uint64_t k12G = UINT64_C(0x300000000);
                                       ^
2 errors generated.
make[2]: *** [lib/system/CMakeFiles/wasmedgeSystem.dir/build.make:76: lib/system/CMakeFiles/wasmedgeSystem.dir/allocator.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:436: lib/system/CMakeFiles/wasmedgeSystem.dir/all] Error 2
```

I get the feeling this code doesn't get built on 32bit systems very often?

I am testing on `arm32v6/alpine`.

```bash
# Install dependencies
apk add cmake ninja clang lld-dev llvm-dev boost-dev musl-dev llvm-static git
export CC=clang
export CXX=clang++
# Get source code
git clone https://github.com/wasmedge/wasmedge
cd wasmedge/
# Configure
cmake -Bbuildmusl -GNinja -DCMAKE_BUILD_TYPE=Release -DWASMEDGE_BUILD_STATIC_LIB=ON -DWASMEDGE_BUILD_SHARED_LIB=OFF -DWASMEDGE_BUILD_AOT_RUNTIME=OFF -DWASMEDGE_BUILD_TOOLS=OFF -Wno-dev .
# Build
cmake --build buildmusl/
```

It will fail due to the following error:

```
ninja: job failed: /usr/bin/clang++ -DSPDLOG_COMPILED_LIB -I/root/wasmedge/buildmusl/lib/system -I/root/wasmedge/buildmusl/include -I/root/wasmedge/include -I/root/wasmedge/buildmusl/_deps/spdlog-src/include -O3 -DNDEBUG -std=c++17 -fPIC -fvisibility=hidden -fvisibility-inlines-hidden -Wall -Wextra -Werror -Wno-error=pedantic -Wno-psabi -Wno-c++20-designator -Wno-c99-extensions -Wno-covered-switch-default -Wno-documentation-unknown-command -Wno-error=nested-anon-types -Wno-error=old-style-cast -Wno-error=unused-command-line-argument -Wno-error=unknown-warning-option -Wno-ctad-maybe-unsupported -Wno-gnu-anonymous-struct -Wno-keyword-macro -Wno-language-extension-token -Wno-newline-eof -Wno-shadow-field-in-constructor -Wno-signed-enum-bitfield -Wno-switch-enum -Wno-undefined-func-template -Wno-error=shadow-field -Wno-reserved-identifier -MD -MT lib/system/CMakeFiles/wasmedgeSystem.dir/allocator.cpp.o -MF lib/system/CMakeFiles/wasmedgeSystem.dir/allocator.cpp.o.d -o lib/system/CMakeFiles/wasmedgeSystem.dir/allocator.cpp.o -c /root/wasmedge/lib/system/allocator.cpp
/root/wasmedge/lib/system/allocator.cpp:25:40: error: unused variable 'k4G' [-Werror,-Wunused-const-variable]
static inline constexpr const uint64_t k4G = UINT64_C(0x100000000);
                                       ^
/root/wasmedge/lib/system/allocator.cpp:26:40: error: unused variable 'k12G' [-Werror,-Wunused-const-variable]
static inline constexpr const uint64_t k12G = UINT64_C(0x300000000);
                                       ^
2 errors generated.
ninja: subcommand failed
```

#2605 will fix it. After I applied #2605, I could build the wasmedge with the above command without any issue.
So we can close this once the PR gets merged.

--- a/lib/system/allocator.cpp
+++ b/lib/system/allocator.cpp
@@ -22,8 +22,14 @@ namespace WasmEdge {
 
 namespace {
 static inline constexpr const uint64_t kPageSize = UINT64_C(65536);
+
+#if WASMEDGE_OS_WINDOWS || defined(HAVE_MMAP) && defined(__x86_64__) ||        \
+    defined(__aarch64__) || (defined(__riscv) && __riscv_xlen == 64)
+// Only define these two constants on the supported platform to avoid
+// -Wunused-const-variable error when applying -Werror.
 static inline constexpr const uint64_t k4G = UINT64_C(0x100000000);
 static inline constexpr const uint64_t k12G = UINT64_C(0x300000000);
+#endif
 
 } // namespace
 

